### ISSUE TITLE
[Feature Request]: Add Sponge support

### ISSUE THREAD

**Issue by ahdg6** [A248/LibertyBans#99]
### I have confirmed that ...

- [X] There is no duplicate feature request.

### Description

For now, Sponge is lacking of awesome Ban-related plugins. ( Sponge is really widely used among modding server, I'm sure there is a strong demand)
LibertyBans will be a powerful competitor to join in, and it will extend the fame of multi-platform supports.

**Comment by A248** [comment]
Could you clarify whether you mean Sponge 7 or Sponge 8?

Quite sadly Sponge 7 requires Java 8, and only Java 8; it forbids the use of later Java versions. As LibertyBans requires Java 11, I can only imagine a Sponge 7 server running LibertyBans.

**Comment by ahdg6** [comment]
oh. I did not realize that. Maybe can keep this open as Sponge-1.16.5 is coming.

**Comment by ahdg6** [comment]
https://forums.spongepowered.org/t/status-update-8th-january-2022-spongeapi-8-released/39861

Sponge API 8 is out.

**Comment by A248** [comment]
Wonderful! I'll add support for it in one of the next two releases.

**Comment by A248** [comment]
Sponge API 9 is released, so that is what we will target.

**Comment by A248** [comment]
We will implement Sponge's BanService if configured, for compatibility with other plugins. For performance purposes it may be disabled by configuration, since the inflexibility of the `BanService#find` methods is insufficient for our needs.

The main punishment check, when a player joins, will happen during the ServerSideConnectionEvent.Handshake event. The BanService will then perform a redundant query if it is enabled; the BanService's purpose is to serve other plugins' needs despite the performance cost.

For more information refer to this conversation over Discord: [Pasted content](https://pastebin.com/dK49fKW4) / [discord link](https://discord.com/channels/142425412096491520/142425521391665153/999842070455926874).

**Comment by A248** [comment]
December 11... to a September 11 release. Perfect.

**Comment by ahdg6** [comment]
would you like to upload libertybans to [ore](ore.spongepowered.org/)? It's a plugin-hosting platform for Sponge and I'm sure Sponge users can get it faster.

**Comment by A248** [comment]
> would you like to upload libertybans to [ore](ore.spongepowered.org/)? It's a plugin-hosting platform for Sponge and I'm sure Sponge users can get it faster.

Fortunate timing with your comment. I intend to upload LibertyBans to Ore.

Currently, however, I am awaiting the resolution of an issue caused by a bug in Ore. [My Ore profile](https://ore.spongepowered.org/A248) is not linked to the [LibertyBans organization](https://ore.spongepowered.org/LibertyBans#) which I created. I've posted a message about this in the Sponge discord and am awaiting a response. Apparently this has happened before, to other users.
